### PR TITLE
DAOS-2487 DFS: Change DFS API to return -errno instead of -DER

### PR DESCRIPTION
**Pull request by mchaarawi**
also update dfuse and dfuse_hl

### CODE REVIEW

File: src/client/dfuse/ops/lookup.c [hunk 1]
```diff
@@ -40,15 +40,15 @@ dfuse_reply_entry(struct dfuse_projection_info *fs_handle,
...
+		if (rc) {
+			D_GOTO(err, rc = -rc);
 		}
 		rc = dfuse_lookup_inode(fs_handle,
 					ie->ie_dfs,
 					&oid,
 					&ie->ie_stat.st_ino);
 		if (rc != -DER_SUCCESS) {
-			D_GOTO(err, rc = EIO);
+			D_GOTO(err, daos_der2errno(rc));
```
[the remark this answers is not in the document]
Comment (mchaarawi) [reply src/client/dfuse/ops/lookup.c]: it actually should be rc = daos_der2errno(rc)
dfuse_lookup_inode() returns -DER. I haven't changed any of the dfuse functions. just the dfs

File: src/client/dfuse/ops/lookup.c [hunk 2]
```diff
@@ -122,7 +122,7 @@ dfuse_cb_lookup(fuse_req_t req, struct dfuse_inode_entry *parent,
 
 	rc = dfs_ostat(parent->ie_dfs->dffs_dfs, ie->ie_obj, &ie->ie_stat);
 	if (rc != -DER_SUCCESS) {
-		D_GOTO(err, 0);
+		D_GOTO(err, rc = -rc);
```
[the remark this answers is not in the document]
Comment (mchaarawi) [reply src/client/dfuse/ops/lookup.c]: well -DER_SUCCESS is 0.. but yea im not sure why we don't just do if (rc)all through the dfuse code.

[the remark this answers is not in the document]
Comment (johannlombardi) [reply src/client/dfuse/ops/lookup.c]: Got it, but if dfs_ functions don't return DER_ error any longer, then it does not make much sense to check the return code against DER_SUCCESS then.

File: src/client/dfuse/dfuse_cont.c [hunk 3]
```diff
@@ -89,10 +88,10 @@ dfuse_cont_open(fuse_req_t req, struct dfuse_inode_entry *parent,
 			 */
 			rc = dfs_ostat(ie->ie_dfs->dffs_dfs,
 				       ie->ie_obj, &entry.attr);
-			if (rc != -DER_SUCCESS) {
+			if (rc) {
```
Comment (johannlombardi) [inline src/client/dfuse/dfuse_cont.c]: here

File: src/client/dfuse/dfuse_cont.c [hunk 4]
```diff
@@ -138,22 +135,18 @@ dfuse_cont_open(fuse_req_t req, struct dfuse_inode_entry *parent,
 
 	rc = dfs_ostat(dfs->dffs_dfs, ie->ie_obj, &ie->ie_stat);
 	if (rc != -DER_SUCCESS) {
-		DFUSE_TRA_ERROR(ie, "dfs_ostat() failed: (%d)",
-				rc);
-		D_GOTO(release, 0);
+		DFUSE_TRA_ERROR(ie, "dfs_ostat() failed: (%s)", strerror(-rc));
```
Comment (johannlombardi) [inline src/client/dfuse/dfuse_cont.c]: Not consistent with above.

File: src/client/dfuse/ops/readdir.c [hunk 5]
```diff
@@ -147,36 +147,36 @@ dfuse_cb_readdir(fuse_req_t req, struct dfuse_inode_entry *inode,
...
+			if (rc) {
 				D_FREE(ie);
-				D_GOTO(err_or_buf, 0);
+				D_GOTO(err_or_buf, rc = -rc);
 			}
 
 			rc = dfs_ostat(inode->ie_dfs->dffs_dfs, ie->ie_obj,
 				       &ie->ie_stat);
-			if (rc != -DER_SUCCESS) {
+			if (rc) {
```
Comment (johannlombardi) [inline src/client/dfuse/ops/readdir.c]: At line 137 in src/client/dfuse/dfuse_cont.c, we check against DER_SUCCESS, we should probably try to be consistent here.

**Pull request merged**